src/components/timers/Countdown.tsx

```diff
@@ -1,3 +1,183 @@
+                if (prev > 1) return prev - 1;
+                return prev; //this is added to fix a TypeScript error to ensure we never return undefined
+            });
+
+            intervalRef.current = setInterval(() => {
+                setSecondsRemaining(prev => {
+                    if (prev > 1) return prev - 1;
+                    else {
+                        setStatus(STATUS.STOPPED);
+                        return 0;
+                    }
+                });
+            }, 1000);
+        }
+        return () => {
+            if (intervalRef.current !== null) {
+                clearInterval(intervalRef.current);
+                intervalRef.current = null;
+            }
+        };
+    }, [status]);
+
+    return (
+        <div className="App">
+            <TimerContainer isActive={status === STATUS.STARTED}>
+                <TimerTitle>Countdown</TimerTitle>
+
+                {/* INPUTS FOR INITIAL STATE*/}
+                <Timer>
+                    {status === STATUS.INITIAL && (
+                        <Inputs>
+                            <Input>
+                                <input
+                                    style={{ maxWidth: '3rem', border: '0px solid white', fontSize: '2rem', textAlign: 'right' }}
+                                    id="timeMinInput"
+                                    placeholder="10"
+                                    value={timeMinInput}
+                                    onChange={e => {
+                                        setTimeMinInput(e.target.value);
+                                    }}
+                                    disabled={secondsRemaining > 0}
+                                />
+                            </Input>
+                            <Input>
+                                :
+                                <input
+                                    style={{ maxWidth: '3rem', border: '0px solid white', fontSize: '2rem', textAlign: 'left' }}
+                                    id="timeInput"
+                                    value={timeSecInput}
+                                    placeholder="00"
+                                    onChange={e => {
+                                        setTimeSecInput(e.target.value);
+                                    }}
+                                    disabled={secondsRemaining > 0}
+                                />
+                            </Input>
+                        </Inputs>
+                    )}
+
+                    {/* TIME DISPLAY*/}
+                    {status !== STATUS.INITIAL && (
+                        <TimeDisplay isActive={status === STATUS.STARTED}>
+                            <DisplayForTime hoursOnTimer={hoursOnTimer} minutesOnTimer={minutesOnTimer} secondsOnTimer={secondsOnTimer} />
+                        </TimeDisplay>
+                    )}
+                </Timer>
+
+                {/* DYMANIC TEXT LINE*/}
```

Comment:
`Countdown` component starts to feel a little busy here once we start getting into `SupportText` and the `Button`.  We're also seeing a lot of the responsibility of sub-components filtering up to this parent component. Why should `Countdown` be responsible for figuring out how `SupportText` displays the text? That's not its job.

You can simplify and DRY things up below by passing in `status` and the rest of the relevant fields as props to `SupportText`. Same thing for the buttons.